odk1-src/form-functions.rst

```diff
@@ -0,0 +1,719 @@
+  
+Converting dates and time
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  
+.. function:: decimal-date-time(dateTime)
+
+  Converts :arg:`dateTime` value to the number of days since January 1, 1970 UTC. This is the format used by Excel.
+  
+  This is the inverse of :func:`date`.
+
+.. function:: date(days)
+
+  Converts an integer representing a number of :arg:`days` from 01 January 1970 (the `Unix Epoch`_) to a standard date value.
+
+  .. _Unix Epoch: https://en.wikipedia.org/wiki/Unix_time
+    
+  This is the inverse of :func:`decimal-date-time`.
+
+    
+.. function:: decimal-time(time)
+
+  Converts :arg:`time` to a number representing a fractional day.
+  For example, noon is 0.5 and 6pm is 0.75.
+
+
+.. _date-time-formatting-functions:
+
+Formatting dates and times for display
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  
+  
+  
+.. function:: format-date(date, format)
+
+  Returns :arg:`date` as a string formatted as defined by :arg:`format`.
+  
+  .. container:: details
+  
+    The following identifiers are used in the :arg:`format` string:
+
+    .. csv-table::
+
+      %Y, 4-digit year
+      %y, 2-digit year
+      %m, 0-padded month
+      %n, numeric month
+      %b, "short text month (Jan, Feb, Mar...)" 
+      %d, 0-padded day of month
+      %e, day of month
+      %a, "short text day (Sun, Mon, Tue...)."
+
+    .. note:: 
+    
+      Month and day abbreviations are language and locale specific. If form locale can be determined, that locale will be used. Otherwise, the device locale will be used.
+  
+.. function:: format-date-time(dateTime, format)
+
+  Returns :arg:`dateTime` as a string formatted as defined by :arg:`format`.
+
+  .. container:: details
+  
+    The identifiers list in :func:`format-date` are available, 
+    plus the following:
+
+    .. csv-table::
+
+      %H, 0-padded hour (24-hr time)
+      %h, hour (24-hr time)
+      %M, 0-padded minute
+      %S, 0-padded second
+      %3, 0-padded millisecond ticks.
+
+.. _geography-functions:
+    
+Geography
+------------
+
+.. function:: area(nodeset | geoshape) 	
+
+  Returns the area, in square meters, 
+  of either a :arg:`nodeset` of geopoints or a :arg:`geoshape` value.
+  
+  It takes into account the circumference of the Earth around the Equator but does not take altitude into account.
+
+.. function:: distance(nodeset | geoshape | geotrace)
+
+  Returns the distance, in meters, of either:
+  
+  - a :arg:`nodeset` of geopoints
+  - the perimeter of a :arg:`geoshape`
+  - the length of a :arg:`geotrace` value
+  
+  It takes into account the circumference of the Earth around the Equator and does not take altitude into account.
+
+.. _utility-functions:
+
+Utility
+---------
+
+.. function:: random()
+
+  Returns a random number between 0.0 (inclusive) and 1.0 (exclusive).
+
+  
+.. function:: randomize(nodeset[, seed]) 	
+
+  Returns a shuffled :arg:`nodeset`.
+  
+  A shuffle with a numeric :arg:`seed` is deterministic and reproducible.
```

Comment:
May want to suggest saving a seed in a previous calculate (e.g. using `random` or `decimal-date-time(now())`) and using it so that the order can be recreated.